### ISSUE TITLE
Breakpoints not reached with remote debugging (WSL2)

### ISSUE THREAD

**Issue by keckelt** [firefox-devtools/vscode-firefox-debug#157]
Hi!
I have Win10 insiders preview with WSL2 and a project inside of the WSL2 ubuntu distro that i want to debug with Firefox Developer.

My assumption is that the paths are not correct because they look like this in my diagnostic logfile:
```
DEBUG|002.453|BreakpointsManager: Setting 1 breakpoints for vscode-remote://wsl%2Bubuntu/home/klaus/ws/cohort-app/cohort/src/index.ts
DEBUG|002.460|BreakpointsManager: Setting 1 breakpoints for vscode-remote://wsl%2Bubuntu/home/klaus/ws/cohort-app/cohort/src/Taskview/SearchColumn.ts

DEBUG|006.909|PathConversion: Converted url webpack:///src/Taskview/SearchColumn.ts to path vscode-remote:\wsl%2Bubuntu\home\klaus\ws\cohort-app\cohort\src\Taskview\SearchColumn.ts
DEBUG|006.910|SkipFilesManager: skipFile is not set for vscode-remote:\wsl%2Bubuntu\home\klaus\ws\cohort-app\cohort\src\Taskview\SearchColumn.ts
```

With `%2B`encoding a `+` this would be `/wsl+ubuntu/...` instead of what Windows shows me: `\\wsl$\ubuntu`.
**Is that an issue?**

Also the path starts with `vscode-remote://...`for the breakpoints but with `vscode-remote:\...` for the mapping.

---

Some more details:
The project is in thefolder `/home/klaus/ws/cohort-app/`, which i open with VS Code from the linux terminal.

In the app folder I have:
```
node_modules/
cohort/   --> my actual app (index.html, index.ts, etc.); main debug target
phovea_*/ --> other repos I need for my app
tdp_*/    --> same
``` 

The project is using TypeScript and is built by webpack. In the devtools debugger tab it looks like this:
![image](https://user-images.githubusercontent.com/10337788/68846531-89e88200-06cd-11ea-99ae-fafd220ed5e8.png)
where `src` is the _src_ directory in the `cohort/` folder.

Based on the description of this extension i created the following launch config in VS Code:
```javascript
{
  "name": "Launch FireFox Dev (rel)",
  "type": "firefox",
  "request": "launch",
  "firefoxExecutable": "C:/Program Files/Firefox Developer Edition/firefox.exe",
  "url": "http://localhost:8080",
  "webRoot": "${workspaceFolder}/cohort/", // because npm start = "cd cohort && npm start"
  "pathMappings": [
    { "url": "webpack:///~/", "path": "${workspaceFolder}/node_modules/" },
    { "url": "webpack:///./~/", "path": "${workspaceFolder}/node_modules/" },
    { "url": "webpack:///./", "path": "${workspaceFolder}/" }, //other folders in my workspace
    { "url": "webpack:///src/", "path": "${webRoot}/src/" }, // src folder in cohort/
    { "url": "webpack:///", "path": "" } // ??
  ],
  "log": {
    "fileName": "C:/Users/Klaus/Desktop/debug-log.txt",
    "fileLevel": {
        "default": "Debug"
    }
  }
}
```

Then I saw the weird paths and rather specified the paths absolute: 
```javascript
{
  "name": "Launch FireFox Dev (absolute)",
  "type": "firefox",
  "request": "launch",
  "firefoxExecutable": "C:/Program Files/Firefox Developer Edition/firefox.exe",
  "url": "http://localhost:8080",
  "webRoot": "/home/klaus/ws/cohort-app/cohort/",
  "pathMappings": [
    { "url": "webpack:///~/", "path": "/home/klaus/ws/cohort-app/node_modules/" },
    { "url": "webpack:///./~/", "path": "/home/klaus/ws/cohort-app/node_modules/" },
    { "url": "webpack:///./", "path": "/home/klaus/ws/cohort-app/" },
    { "url": "webpack:///src/", "path": "${webRoot}/src/" },
    { "url": "webpack:///", "path": "" }
  ],
  "log": {
    "fileName": "C:/Users/Klaus/Desktop/debug-log-abs.txt",
    "fileLevel": {
        "default": "Debug"
    }
  }
}
```

Then the paths printed by _PathConversion_ look fine, but the BreakpointsManager paths are still the same and breakpoints can not be reached:
```
DEBUG|002.442|BreakpointsManager: Setting 1 breakpoints for vscode-remote://wsl%2Bubuntu/home/klaus/ws/cohort-app/cohort/src/index.ts
DEBUG|002.443|BreakpointsManager: Setting 1 breakpoints for vscode-remote://wsl%2Bubuntu/home/klaus/ws/cohort-app/cohort/src/Taskview/SearchColumn.ts

DEBUG|007.915|PathConversion: Converted url webpack:///src/Taskview/SearchColumn.ts to path \home\klaus\ws\cohort-app\cohort\src\Taskview\SearchColumn.ts
DEBUG|007.916|SkipFilesManager: skipFile is not set for \home\klaus\ws\cohort-app\cohort\src\Taskview\SearchColumn.ts
```

---

I'm able to debug in chrome with this config:
```javascript
 {
  "name": "Launch Chrome",
  "type": "chrome",
  "request": "launch",
  "userDataDir": true, // (optional); start debugging in new window. Enabled by default, but disabled by the use of runtimeExecutable. 
  "url": "http://localhost:8080",
  "webRoot": "${workspaceFolder}/cohort/", // because npm start = "cd cohort && npm start"
  "sourceMapPathOverrides": { // help the vs code debugger finding the sources. 
    "webpack:///./*": "${webRoot}/*",  // '.'' for cohort code (the current app, where webpack is started) 
    "webpack:///../*": "${workspaceFolder}/*",  // '..' for the other modules. The debugger does not infer that from the mapping of '.'
  }
},
```

There, the webpack sources look a bit different:
![image](https://user-images.githubusercontent.com/10337788/68848649-ee591080-06d0-11ea-8ad1-bfc5c3090984.png)

**Comment by hbenl** [comment]
Thanks for the detailed description. The problem is probably the mismatch between `vscode-remote://...` and `vscode-remote:\...`. Apparently the extension treats the `vscode-remote` URLs like paths and normalizes `/` into `\`. I will hopefully be able to fix this over the weekend.

**Comment by keckelt** [comment]
Great, thanks for looking into it!
I'm happy to try any changes.

**Comment by hbenl** [comment]
Fixed in version 2.5.2

**Comment by keckelt** [comment]
Thank you!
I updated and it works now 🙌


Just had two more errors in my config:
1. (trailing slash in webRoot definition) plus a second slash in the pathmapping.
2. I confused the mapping of `` with what `` does.

This is the config I'm using now:
```js
{
  "name": "Launch FireFox Dev",
  "type": "firefox",
  "request": "launch",
  "url": "http://localhost:8080",
  "webRoot": "${workspaceFolder}/cohort", // because npm start = "cd cohort && npm start"
  "pathMappings": [
    { "url": "webpack:///~/", "path": "${workspaceFolder}/node_modules/" },
    { "url": "webpack:///./~/", "path": "${workspaceFolder}/node_modules/" },
    { "url": "webpack:///./", "path": "${webRoot}/" }, // probably files directly in my workspace/cohort folder
    { "url": "webpack:///src/", "path": "${webRoot}/src/" }, // src folder in workspace/cohort/
    { "url": "webpack:///", "path": "${workspaceFolder}/" } // other folders in my workspace
  ],
  "log": {
    "fileName": "C:/Users/Klaus Eckelt/Desktop/debug-log.txt",
    "fileLevel": {
        "default": "Debug"
    }
  }
}
```

It automatically picked the Developer version of Firefox :)
